File: python/src/sdk.py
from .generated.swagger_petstore_open_api_3_0_client import Client as Client
from .generated.swagger_petstore_open_api_3_0_client.api.pet import get_pet_by_id
from .generated.swagger_petstore_open_api_3_0_client.models import Pet


class Acme:
    """
    A python SDK for the Petstore API. It uses the generated client to perform requests.
    Like the other SDKs, it provides a high-level interface with type bindings.
    """

    def __init__(self, base_url: str):
        self.client = Client(base_url=base_url)

    async def get_pet(self, pet_id) -> Pet:
        async with self.client as client:
            model: Pet = await get_pet_by_id.asyncio(client=client, pet_id=pet_id)
            return model

File: javascript/src/sdk.ts
import { getPetById, Pet } from "./generated";
import { Client, createClient } from '@hey-api/client-fetch';

import { Config } from "@hey-api/client-fetch";

/**
 * This is an example of how to use the generated client to build a higher-level SDK.
 * Notice that the code here is fully type-safe, and we don't have to manually maintain
 * the API type bindings - the openapi library takes care of that.
 */
export class Acme {

    /**
     * The underlying HTTP client that performs the requests.
     */
    readonly client: Client;

    constructor(config: Config) {
        this.client = createClient(config);

        // Register some middleware
        this.client.interceptors.request.use((request) => {
            request.headers.set('content-type', 'application/json');
            // We can add more logic here, e.g. auth, analytics, etc.
            return request;
        });
    }

    async getPetById(petId: number): Promise<Pet | undefined> {
        const response = await getPetById({
            client: this.client,
            path: {
                petId,
            },
        });

        if (response.error) {
            throw new Error(`Failed to get pet by ID: ${JSON.stringify(response.error)}`);
        }

        return response.data;
    }
}
